app/src/main/java/com/github/bgabriel998/softwaredevproject/CameraPreview.java

```diff
@@ -2,63 +2,173 @@
+ * Use the {@link CameraPreview#newInstance} factory method to
+ * create an instance of this fragment.
+ * See: https://github.com/android/camera-samples/blob/main/CameraXBasic
  */
-public class CameraPreview{
+public class CameraPreview extends Fragment{
 
-    private final PreviewView previewView;
+    private PreviewView previewView;
     private ProcessCameraProvider cameraProvider;
-    private final ExecutorService cameraExecutor;
-    private final Context context;
+    private ImageCapture imageCapture;
+    private ExecutorService cameraExecutor;
+    private Context context;
+
+    private final String FILENAME = "yyyy-MM-dd-HH-mm-ss-SSS";
+    private final String PHOTO_EXTENSION = ".jpg";
+    private final int PICTURE_TAKEN = 1;
+    private final int FAILED_TO_TAKE_PICTURE = 0;
+    private final int FILE_LENGTH = 27;
+
+    private ImageAnalysis imageAnalysis;
+
+    private int previewDisplayId = -1;
+
+    private DisplayManager displayManager;
+
+    private DisplayManager.DisplayListener displayListener;
+
+    String lastToast = null;
 
     /**
      * Constructor for the CameraPreview
-     * @param context Context of the activity on which the camera-preview is drawn
-     * @param previewView View on which the camera-preview is binded to
+     * Is required to be empty for the fragments
+     */
+    public CameraPreview() {}
+
+    /**
+     * Use this factory method to create a new instance of
+     * this fragment using the provided parameters.
+     *
+     * @return A new instance of fragment CameraPreview.
      */
-    public CameraPreview(Context context, PreviewView previewView){
-        this.context = context;
-        this.previewView = previewView;
+    public static CameraPreview newInstance() {
+        return new CameraPreview();
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        return inflater.inflate(R.layout.camera_preview, container, false);
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        ConstraintLayout container = (ConstraintLayout) view;
+        previewView = container.findViewById(R.id.cameraPreview);
 
         //Initialize background executor
         cameraExecutor = Executors.newSingleThreadExecutor();
 
-        //Wait for the view to be properly laid out and then setup the camera
-        this.previewView.post(this::setUpCamera);
+        //Register display listener
+        displayManager.registerDisplayListener(displayListener, null);
+
+        //Get outputdirectory to store image
+        //outputDirectory = Button1Activity.getOutputDirectory(requireContext());
```

Comment:
What is this?

Reply:
This was some leftover code that I used to store the image. But this is handled in another location in code. I forgot to remove this.